From: Hans-Joachim Picht <hpicht@redhat.com>
Date: Thu, 12 Mar 2009 15:26:30 +0100
Subject: [s390] set default preferred console device 'ttyS'
Message-id: 20090312142630.GP5103@redhat.com
O-Subject: [RHEL5 U4 PATCH 15/20] FEAT: s390 - set default preferred console device "ttyS"
Bugzilla: 475551

Description:
============

This patch explicitly sets the preferred console device to ttyS0.
This is a pre-req for the z/VM IUCV HVC device driver.

The console= kernel parameter can be still used to switch the preferred
console to some other device. In that case, console messages are also
printed on the default console device (ttyS0).

Bugzilla:
============
BZ  475551
https://bugzilla.redhat.com/show_bug.cgi?id=475551

Upstream status of the patch:
==============================

The patch is upstream as git commit
0946100f566d04cc93e86654f3ba588ee94b3f4c

Test status:
============

The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Please ACK.

With best regards,

	--Hans

diff --git a/arch/s390/kernel/setup.c b/arch/s390/kernel/setup.c
index 5cd2733..bf01463 100644
--- a/arch/s390/kernel/setup.c
+++ b/arch/s390/kernel/setup.c
@@ -896,6 +896,9 @@ static void __init setup_hwcaps(void)
 void __init
 setup_arch(char **cmdline_p)
 {
+	/* set up preferred console */
+	add_preferred_console("ttyS", 0, NULL);
+
         /*
          * print what head.S has found out about the machine
          */
